CriteoPublisherSdk/Sources/CR_BidManager.m

```diff
@@ -179,6 +180,47 @@ - (void)prefetchBidsForAdUnits:(CR_CacheAdUnitArray *)adUnits {
          }];
 }
 
+- (void)fetchLiveBidForAdUnit:(CR_CacheAdUnit *)adUnit
+           bidResponseHandler:(CR_BidResponseHandler)responseHandler {
+  [self fetchLiveBidForAdUnit:adUnit
+           bidResponseHandler:responseHandler
+                   timeBudget:CRITEO_DEFAULT_LIVE_BID_TIME_BUDGET_IN_SECONDS];
+}
+
+- (void)fetchLiveBidForAdUnit:(CR_CacheAdUnit *)adUnit
+           bidResponseHandler:(CR_BidResponseHandler)responseHandler
+                   timeBudget:(NSTimeInterval)timeBudget {
+  [self.threadManager dispatchAsyncOnGlobalQueueWithTimeout:timeBudget
+      operationHandler:^void(void (^completionHandler)(dispatchWithTimeoutHandler)) {
+        [self fetchBidsForAdUnits:@[ adUnit ]
+               cdbResponseHandler:^(CR_CdbResponse *cdbResponse) {
+                 completionHandler(^(BOOL handled) {
+                   [self.threadManager dispatchAsyncOnMainQueue:^{
```

Comment:
Why do you need to go on main queue ?

Reply:
Because most callbacks (Standalone) are interacting with UI, which requires to be on main queue.

That could be higher up to `Criteo` later thought, is that your point?

Reply:
Yep this could be done later by client code if they need to (for example header bidding does not need the main thread). Jumping on the main thread here seems just surprising.
Moving this logic to caller is FAR. In the meantime, maybe could you just add a comment indicating the why ?

Reply:
Fixed by aaeb53f